hooks/new

```diff
@@ -0,0 +1,383 @@
+		'What is your Azure Client Secret?'
+	prompt_for azure_tenant_id line \
+		'What is your Azure Tenant ID?'
+	prompt_for azure_subscription_id line \
+		'What is your Azure Subscription ID?'
+	prompt_for azure_resource_group line \
+		'What Azure Resource Group will BOSH be deploying VMs into?'
+	safe --quiet set secret/$prefix/azure client_id="$azure_client_id" \
+	                                      tenant_id="$azure_tenant_id" \
+	                                      subscription_id="$azure_subscription_id"
+
+	prompt_for azure_default_sg line \
+		'What security group should be used as the BOSH default security group?'
+
+	if [[ $isproto == "true" ]]; then
+		prompt_for azure_vnet line \
+			'What is the name of your Azure Virtual Network?'
+		prompt_for azure_subnet_name line \
+			'What is the name of the Azure subnet that the BOSH will be placed in?'
+	fi
+	;;
+
+openstack)
+	prompt_for openstack_auth_url line \
+		'What is the Auth URL of your OpenStack cluster?'
+	prompt_for openstack_user line \
+		'What username will be used to authenticate with OpenStack?'
+	prompt_for $prefix/openstack/creds:password secret-line \
+		'What password will be used to authenticate with OpenStack?'
+	prompt_for openstack_domain line \
+		'What OpenStack Domain will BOSH be deployed in?'
+	prompt_for openstack_project line \
+		'What OpenStack Project will BOSH be deployed in?'
+	safe set --quiet secret/$prefix/openstack/creds username="$openstack_user" \
+	                                                domain="$openstack_domain" \
+	                                                project="$openstack_project"
+
+	prompt_for openstack_region line \
+		'What OpenStack Region is BOSH being deployed to?'
+	prompt_for openstack_ssh_key line \
+		'What is the name of the OpenStack SSH key that should be used to enable SSH access to BOSH-deployed VMs?'
+	prompt_for openstack_default_sgs multi-line \
+		'What default security groups should be applied to VMs created by BOSH?'
+
+	if [[ $isproto == "true" ]]; then
+		prompt_for openstack_network_id line \
+			'What is the UUID of the OpenStack network that BOSH will be placed in?'
+		prompt_for openstack_flavor line \
+			'What OpenStack flavor (instance type) should the BOSH VM use?'
+		prompt_for openstack_az line \
+			'What AZ will the BOSH Director be placed in?'
+	fi
+	;;
+esac
+
+###########################################################################
+
+(
+cat <<EOF
+kit:
+  features:
+    - $iaas
+EOF
+if [[ $isproto == 'true' ]]; then
+	echo "    - proto"
+fi
+
+cat <<EOF
+
+params:
+  env:   $name
+  vault: $prefix
+
+EOF
+
+if [[ $isproto == 'true' ]]; then
+cat <<EOF
+  # These properties definte the host-level networking for a proto-BOSH.
+  # Environmental BOSHes can depend on their parent BOSH cloud-config,
+  # but for proto- environments, we have to specify these.
+  #
+  static_ip:       $ip
+  subnet_addr:     $proto_net
+  default_gateway: $proto_gw
+  dns:
+EOF
```

Comment:
Dennis and I discussed this and if we're going to start a cargo cult with this kit and its implementation, we had better do a dang good job and build a solid foundational religion to cult out.

Remove the heredocs and use double-quoted echos or no-arg echos (for empty lines) instead.